test/unit/host-config.spec.ts

```diff
@@ -0,0 +1,701 @@
+                    redirector = foo;
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({});
+		});
+
+		it('parses to empty base config while some values are invalid', () => {
+			// Only `redirector` is invalid
+			const redsocksConfStr = stripIndent`
+                base {
+                    log_debug = off;
+                    log_info = on;
+                    log = stderr;
+                    daemon = off;
+                    redirector = foo;
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({});
+		});
+
+		it('parses `redsocks {...}` config block', () => {
+			const redsocksConfStr = stripIndent`
+                redsocks {
+                    local_ip = 127.0.0.1;
+                    local_port = 12345;
+                    type = http-connect;
+                    ip = test2.balenadev.io;
+                    port = 1082;
+                    login = "user";
+                    password = "pass";
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({
+				redsocks: {
+					local_ip: '127.0.0.1',
+					local_port: 12345,
+					type: 'http-connect',
+					ip: 'test2.balenadev.io',
+					port: 1082,
+					login: '"user"',
+					password: '"pass"',
+				},
+			});
+		});
+
+		// Allow invalid fields through for backwards compatibility
+		it('parses to redsocks config with warnings while any values are invalid', () => {
+			const redsocksConfStr = stripIndent`
+                redsocks {
+                    local_ip = 123;
+                    local_port = foo;
+                    type = socks6;
+                    ip = 456;
+                    port = bar;
+                    login = user;
+                    password = pass;
+                }
+            `;
+			(log.warn as SinonStub).resetHistory();
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect((log.warn as SinonStub).lastCall.args[0]).to.equal(
+				'Unexpected redsocks.conf configuration detected, proxy may not behave as expected:\n' +
+					'Expecting NumericIdentifier at 0.0.port but instead got: "bar" (must be be an positive integer)\n' +
+					'Expecting one of:\n' +
+					'    "socks4"\n' +
+					'    "socks5"\n' +
+					'    "http-connect"\n' +
+					'    "http-relay"\n' +
+					'at 0.0.type but instead got: "socks6"\n' +
+					'Expecting DoubleQuotedString at 0.1.login but instead got: <REDACTED> (must be enclosed in double quotes)\n' +
+					'Expecting DoubleQuotedString at 0.1.password but instead got: <REDACTED> (must be enclosed in double quotes)\n' +
+					'Expecting NumericIdentifier at 1.local_port but instead got: "foo" (must be be an positive integer)',
+			);
+			(log.warn as SinonStub).resetHistory();
+			expect(conf).to.deep.equal({
+				redsocks: {
+					local_ip: '123',
+					local_port: 'foo',
+					type: 'socks6',
```

Comment:
I know the API endpoint acts as a patch, but I wonder if we should use a default instead of maintaining the bad configuration

Reply:
For backwards compatibility, the warning would be enough, combined with the fact that the redsocks service would be error looping due to the bad value